python/sdist/amici/numpy.py

```diff
@@ -238,6 +238,25 @@ def __getitem__(self, item: str) -> Union[np.ndarray, ReturnDataPtr,
             item = 'ts'
         return super(ReturnDataView, self).__getitem__(item)
 
+    def by_id(self, model: Model, field: str, entity_id: str) -> np.array:
+        """
+        Get the value of a given field for a named entity.
+
+        :param model: The model from which this ReturnDataView was generated.
+        :param field: The requested field, e.g. 'x' for model states
+        :param entity_id: The ID of the model entity that is to be extracted
+            from ``field`` (e.g. a state ID).
+        """
+        if field in {'x', 'x0', 'sx'}:
+            col_index = model.getStateIds().index(entity_id)
+        elif field in {'w'}:
+            col_index = model.getExpressionIds().index(entity_id)
```

Comment:
`y`/`sy` and `model.getObservableIds()`?

Reply:
what about only assigning ids in the if clause, e.g.,
```
ids = model.getExpressionIds()
```
and then doing the indexing outside:
```
col_index = ids.index(entity_id)
```
we are fine with the standard error message if `entity_id` is not a valid id, right?